src/partition/repart.c

```diff
@@ -3497,12 +3551,16 @@ static int context_mkfs(Context *context) {
                 if (r < 0)
                         return r;
 
+                r = make_copy_files_denylist(context, p, &denylist);
+                if (r < 0)
+                        return r;
+
```

Comment:
if you do what i suggested in my earlier comment, then you can compile a global denylist once, instead of a new one for each call, because then we'll ensure to always copy at least the top-level referenced inode, just skip the children.